src/input_parsers/OnnxParser.cpp

```diff
@@ -0,0 +1,1265 @@
+        if (names.exists(name))
+        {
+            String errorMessage = "nodes in Onnx network must have a unique name";
+            throw MarabouError(MarabouError::ONNX_PARSE_ERROR, errorMessage.ascii());
+        }
+        else
+        {
+            _nodeMap.insert(name, &node);
+            names.insert(name);
+        }
+    }
+
+    for (const onnx::TensorProto& initializer : _network.initializer())
+    {
+        const std::string name = initializer.name();
+        if (names.exists(name))
+        {
+            String errorMessage = "nodes in Onnx network must have a unique name";
+            throw MarabouError(MarabouError::ONNX_PARSE_ERROR, errorMessage.ascii());
+        }
+        else
+        {
+            _constantMap.insert(name, &initializer);
+            names.insert(name);
+        }
+    }
+}
+
+void OnnxParser::validateUserInputNames(Set<String>& inputNames)
+{
+    // Validate the provided inputs
+    for (String inputName : inputNames)
+    {
+        if ( !_nodeMap.exists(inputName) )
+        {
+            String errorMessage = Stringf("Input %s not found in graph!", inputName.ascii());
+            throw MarabouError( MarabouError::ONNX_PARSE_ERROR, errorMessage.ascii() );
+        }
+    }
+}
+
+void OnnxParser::validateUserOutputNames(String &outputName)
+{
+    for (auto node : _network.node())
+    {
+        for (String outputNodeName : node.output())
+        {
+            if (outputName == outputNodeName)
+            {
+                return;
+            }
+        }
+    }
+
+    String errorMessage = Stringf("Output %s not found in graph!", outputName.ascii());
+    throw MarabouError( MarabouError::ONNX_PARSE_ERROR, errorMessage.ascii() );
+}
+
+Set<String> OnnxParser::readInputNames()
+{
+    assert(_network.input().size() >= 1);
+
+    Set<String> initializerNames;
+    for (auto &initNode : _network.initializer())
+    {
+        std::cout << "\ninitialiser: " + initNode.name();
+        initializerNames.insert(initNode.name());
+    }
+
+    Set<String> inputNames;
+    for (auto &inputNode : _network.input())
+    {
+        std::cout << "\ninput: " + inputNode.name();
+        inputNames.insert(inputNode.name());
+    }
+
+    return Set<String>::difference(inputNames, initializerNames);
+}
+
+String OnnxParser::readOutputName()
+{
+    if (_network.output().size() > 1)
+    {
+        String message = "Your model has multiple outputs defined\n";
+        message += "Please specify the name of the output you want to consider using the 'outputName' argument\n";
+        message += "Possible options:";
+        for (auto output : _network.output())
+        {
+            message += " " + output.name();
+        }
+        throw MarabouError(MarabouError::ONNX_PARSE_ERROR, message.ascii());
+    }
+    std::cout << "\noutput: " + _network.output()[0].name();
+    return _network.output()[0].name();
+}
+
+void OnnxParser::initialiseShapeMap()
```

Comment:
Can we use initialize here instead, for consistency with the rest of the code base?

Reply:
Changed to `z` everywhere